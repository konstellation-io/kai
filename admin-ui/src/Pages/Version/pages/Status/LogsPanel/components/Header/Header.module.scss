@import 'Styles/mixins';
@import 'Styles/funcs';
@import 'Styles/colors';

.container {
  display: flex;
  align-items: center;

  transition: height ease 0.3s;

  &.opened {
    height: 4 * $grid-unit;
    flex-shrink: 0;

    .buttons {
      opacity: 1;
    }

    .title {
      pointer-events: none;
    }
  }
}

.title {
  @include font-caption;
  color: font-color(lowlight);

  display: flex;
  align-items: center;
  margin: 0 20px 0 20px;
  transition: color ease 0.3s;
  cursor: pointer;

  svg {
    margin: 4px 1.5 * $grid-unit 0 0;
  }

  &:hover {
    color: font-color(light);
  }
}

.runtime {
  @include font-body;
  color: font-color(light);
}

.version {
  @include font-body;
  color: font-color(light);

  margin-left: 16px;
}

.buttons {
  display: flex;
  flex-direction: row;
  transition: opacity ease 0.4s;
  margin: 4px 8px 0 auto;

  div {
    margin-left: 8px;
    cursor: pointer;

    &:hover {
      color: font-color(light);
    }
  }
}

.stickBottom {
  transition: color ease 0.4s, background-color ease 0.4s;

  &.active {
    color: font-color(light);
    background-color: $bg-color-lighter;

    &:hover {
      color: font-color(light);
    }
  }
}
