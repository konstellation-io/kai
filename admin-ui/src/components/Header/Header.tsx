import { get } from 'lodash';

import React, { FunctionComponent, ReactElement } from 'react';

import Settings from '../../components/Settings/Settings';

import { loader } from 'graphql.macro';
import { useQuery } from '@apollo/react-hooks';

import { GetUserEmail } from '../../graphql/queries/types/GetUserEmail';

import styles from './Header.module.scss';

const GetUserEmailQuery = loader('../../graphql/queries/getUserEmail.graphql');

type Props = {
  children?: ReactElement | ReactElement[] | null;
};
const Header: FunctionComponent<Props> = ({ children }) => {
  const { data, loading } = useQuery<GetUserEmail>(GetUserEmailQuery);

  if (loading)
    return <div className={styles.splash} data-testid="splashscreen" />;

  const username: string = get(data, 'me.email');

  return (
    <header className={styles.container} data-testid="app-header">
      <img
        className={styles.konstellationsIcon}
        src={'/img/brand/konstellation.png'}
        alt="konstellation logo"
      />
      <div className={styles.customHeaderElements}>{children}</div>
      <Settings label={username} />
    </header>
  );
};

export default Header;
