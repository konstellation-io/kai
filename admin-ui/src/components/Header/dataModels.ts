import gql from 'graphql-tag';

export const GET_USERNAME = gql`
  query GetUserName {
    me {
      email
    }
  }
`;
