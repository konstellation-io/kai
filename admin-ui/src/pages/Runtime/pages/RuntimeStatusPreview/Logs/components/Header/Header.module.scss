@import '../../../../../../../styles/mixins';
@import '../../../../../../../styles/funcs';
@import '../../../../../../../styles/colors';

.container {
  display: flex;
  align-items: center;
}

.statusCircle {
  width: 8px;
  height: 8px;
  margin: 0 16px;
  border-radius: 50%;

  @each $status in 'PUBLISHED' {
    background-color: get-color-prop($status);
  }
}

.runtime {
  @include font-body;
  color: $font-color-white;
}

.version {
  @include font-body;
  color: $font-color-regular;

  margin-left: 16px;
}

.buttons {
  display: flex;
  flex-direction: row;
  opacity: 0;
  transition: opacity ease 0.4s;
  margin: 4px 8px 0 auto;

  &.opened {
    opacity: 1;
  }

  div {
    margin-left: 8px;
    cursor: pointer;

    &:hover {
      color: $font-color-white;
    }
  }
}

.stickBottom.active {
  color: $font-color-white;

  &:hover {
    color: $font-color-light;
  }
}
