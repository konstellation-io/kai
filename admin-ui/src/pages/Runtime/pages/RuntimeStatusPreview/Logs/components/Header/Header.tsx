import React from 'react';

import IconClose from '@material-ui/icons/Close';
import IconStickBottom from '@material-ui/icons/VerticalAlignBottom';

import cx from 'classnames';
import styles from './Header.module.scss';

type Props = {
  runtimeName: string;
  versionName: string;
  closeLogs: () => void;
  opened: boolean;
  stickToBottom: boolean;
  toggleStickToBottom: () => void;
};
function Header({
  runtimeName,
  versionName,
  closeLogs,
  opened,
  stickToBottom,
  toggleStickToBottom
}: Props) {
  function scrollToBottom() {
    const listContainer = document.getElementById('VersionLogsListContainer');
    if (listContainer) {
      listContainer.scrollTo({
        top: listContainer.scrollHeight,
        behavior: 'smooth'
      });
    }
  }

  return (
    <div className={styles.container}>
      <div className={styles.statusCircle} />
      <span className={styles.runtime}>{runtimeName}</span>
      <span className={styles.version}>{versionName}</span>
      <div
        className={cx(styles.buttons, {
          [styles.opened]: opened
        })}
      >
        <div
          className={cx(styles.stickBottom, { [styles.active]: stickToBottom })}
          onClick={toggleStickToBottom}
        >
          <IconStickBottom className="icon-regular" />
        </div>
        <div onClick={closeLogs}>
          <IconClose className="icon-regular" />
        </div>
      </div>
    </div>
  );
}

export default Header;
