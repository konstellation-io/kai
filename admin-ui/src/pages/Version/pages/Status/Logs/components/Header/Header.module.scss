@import '../../../../../../../styles/mixins';
@import '../../../../../../../styles/funcs';
@import '../../../../../../../styles/colors';

.container {
  display: flex;
  align-items: center;

  transition: height ease 0.3s;

  &.opened {
    height: 50px;
    flex-shrink: 0;

    .buttons {
      opacity: 1;
    }
  }
}

.title {
  @include font-caption;
  color: $font-color-light;

  display: flex;
  align-items: center;
  margin: 0 20px 0 20px;

  span {
    font-size: 14px;
  }

  svg {
    margin-right: 8px;
  }
}

.runtime {
  @include font-body;
  color: $font-color-white;
}

.version {
  @include font-body;
  color: $font-color-light;

  margin-left: 16px;
}

.buttons {
  display: flex;
  flex-direction: row;
  transition: opacity ease 0.4s;
  margin: 4px 8px 0 auto;

  div {
    margin-left: 8px;
    cursor: pointer;

    &:hover {
      color: $font-color-white;
    }
  }
}

.stickBottom {
  transition: color ease 0.4s, background-color ease 0.4s;

  &.active {
    color: $font-color-white;
    background-color: $bg-color-lighter;

    &:hover {
      color: $font-color-light;
    }
  }
}
