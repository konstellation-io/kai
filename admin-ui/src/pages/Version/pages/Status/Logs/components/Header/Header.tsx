import React from 'react';

import IconOpen from '@material-ui/icons/ExpandLess';
import IconClose from '@material-ui/icons/ExpandMore';
import IconStickBottom from '@material-ui/icons/VerticalAlignBottom';
import IconClear from '@material-ui/icons/DeleteOutline';
import IconLogs from '@material-ui/icons/ListAlt';

import { useApolloClient } from '@apollo/react-hooks';

import cx from 'classnames';
import styles from './Header.module.scss';

type Props = {
  togglePanel: () => void;
  opened: boolean;
  stickToBottom: boolean;
  toggleStickToBottom: () => void;
};
function Header({
  togglePanel,
  opened,
  stickToBottom,
  toggleStickToBottom
}: Props) {
  const client = useApolloClient();

  function clearLogs(): void {
    client.writeData({ data: { logs: [] } });
  }

  const Icon = opened ? IconClose : IconOpen;

  return (
    <div
      className={cx(styles.container, {
        [styles.opened]: opened
      })}
    >
      <div className={styles.title}>
        <IconLogs className="icon-regular" />
        <span>Logs console</span>
      </div>
      <div className={styles.buttons}>
        <div onClick={clearLogs}>
          <IconClear className="icon-regular" />
        </div>
        <div
          className={cx(styles.stickBottom, { [styles.active]: stickToBottom })}
          onClick={toggleStickToBottom}
        >
          <IconStickBottom className="icon-regular" />
        </div>
        <div onClick={togglePanel}>
          <Icon className="icon-regular" />
        </div>
      </div>
    </div>
  );
}

export default Header;
