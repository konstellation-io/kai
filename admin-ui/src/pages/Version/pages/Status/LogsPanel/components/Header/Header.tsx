import React from 'react';

import IconOpen from '@material-ui/icons/ExpandLess';
import IconClose from '@material-ui/icons/ExpandMore';
import * as ICONS from '../../../../../../../constants/icons';

import cx from 'classnames';
import styles from './Header.module.scss';
import SvgIcon from '@material-ui/core/SvgIcon';

type Props = {
  togglePanel: () => void;
  opened: boolean;
};
function Header({ togglePanel, opened }: Props) {
  const Icon = opened ? IconClose : IconOpen;

  return (
    <div
      className={cx(styles.container, {
        [styles.opened]: opened
      })}
    >
      <div className={styles.title}>
        <SvgIcon className="icon-small">
          <path d={ICONS.TERMINAL} />
        </SvgIcon>
        <span>LOGS CONSOLE</span>
      </div>
      <div className={styles.buttons}>
        <div onClick={togglePanel}>
          <Icon className="icon-regular" />
        </div>
      </div>
    </div>
  );
}

export default Header;
