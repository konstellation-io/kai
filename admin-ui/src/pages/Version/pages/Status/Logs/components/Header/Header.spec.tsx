import React from 'react';
import Header from './Header';

import IconClose from '@material-ui/icons/Close';
import IconStickBottom from '@material-ui/icons/VerticalAlignBottom';
import IconClear from '@material-ui/icons/DeleteOutline';
import { shallow, ShallowWrapper } from 'enzyme';

const mockWriteData = jest.fn();
jest.mock('@apollo/react-hooks', () => ({
  useApolloClient: () => ({
    writeData: mockWriteData
  })
}));

describe('Logs Header', () => {
  let wrapper: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;
  const closeLogsMock = jest.fn();
  const toggleStickToBottomMock = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <Header
        closeLogs={closeLogsMock}
        opened
        stickToBottom
        toggleStickToBottom={toggleStickToBottomMock}
      />
    );
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('show right components', () => {
    expect(wrapper.find('.buttons > div').length).toBe(3);
  });

  it('handles events', () => {
    wrapper
      .find(IconClose)
      .parent()
      .simulate('click');
    wrapper
      .find(IconStickBottom)
      .parent()
      .simulate('click');
    wrapper
      .find(IconClear)
      .parent()
      .simulate('click');

    expect(closeLogsMock).toHaveBeenCalledTimes(1);
    expect(toggleStickToBottomMock).toHaveBeenCalledTimes(1);
    expect(mockWriteData).toHaveBeenCalledTimes(1);
  });
});
