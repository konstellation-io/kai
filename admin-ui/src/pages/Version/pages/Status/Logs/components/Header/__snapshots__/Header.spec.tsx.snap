// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logs Header matches snapshot 1`] = `
<div
  className="container opened"
>
  <div
    className="title"
  >
    <ListAltIcon
      className="icon-regular"
    />
    <span>
      Logs console
    </span>
  </div>
  <div
    className="buttons"
  >
    <div
      onClick={[Function]}
    >
      <DeleteOutlineIcon
        className="icon-regular"
      />
    </div>
    <div
      className="stickBottom active"
      onClick={[MockFunction]}
    >
      <VerticalAlignBottomIcon
        className="icon-regular"
      />
    </div>
    <div
      onClick={[MockFunction]}
    >
      <CloseIcon
        className="icon-regular"
      />
    </div>
  </div>
</div>
`;
